### PR TITLE
feat: upgradable showcase

### PR DESCRIPTION
**Pull request by Foivos**
[AXE-4735]

[AXE-4735]: https://axelarnetwork.atlassian.net/browse/AXE-4735?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: move/upgradable/v3/upgradable/sources/upgradable.move [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        let data = DataV1 {
+            value,
+            version: VERSION,
+            version_control,
+            tag: ascii::string(b""),
+        };
+        dynamic_field::add(&mut self.id, 0, data);
+    }
+
+    fun borrow_data(self: &Singleton): &DataV1 {
```
Comment (milapsheth) [inline move/upgradable/v3/upgradable/sources/upgradable.move:61]: Can the index syntax be used to simplify this?
https://move-book.com/reference/index-syntax.html

File: move/upgradable/v3/upgradable/sources/upgradable.move [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            ],
+        ]);
+
+        let data = DataV1 {
+            value,
+            version: VERSION,
+            version_control,
+            tag: ascii::string(b""),
+        };
+        dynamic_field::add(&mut self.id, 0, data);
```
[the remark this answers is not in the document]
Comment (Foivos) [reply move/upgradable/v3/upgradable/sources/upgradable.move:58]: If we just destroy the `DataV1` object, which is what would happen upon such an upgrade, then the function would indeed stop working, but there are no 'overloaded' functions, we would need to define a new one (and keep the old one that does nothing) that takes in `DataV2`. Also, managing multiple different `VersionControl` objects that each need a permissioning scheme to be upgraded themselves, seems like a bad decision.

File: move/upgradable/version_control/sources/version_control.move [hunk 3]
```diff
@@ -0,0 +1,35 @@
+module version_control::version_control {
+    const EVersionNotSupported: u64 = 0;
+
+    public struct VersionControl has copy, store, drop {
+        data: vector<u64>,
```
[the remark this answers is not in the document]
Comment (Foivos) [reply move/upgradable/version_control/sources/version_control.move:5]: yes it should

File: move/upgradable/v2/upgradable/sources/upgradable.move [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    public struct ValueSet has copy, drop {
+        value: u64,
+    }
+
+    entry fun upgrade(self: &mut Singleton) {
+        let data = self.borrow_data_mut();
+        assert!(data.version < VERSION, EIncorrectVersion);
+        data.version = VERSION;
+        data.version_control = version_control::new(vector[
```
[the remark this answers is not in the document]
Comment (Foivos) [reply move/upgradable/v2/upgradable/sources/upgradable.move:32]: that's just gravy but sure

**Pull request closed without merging**